sys/kern/subr_bus.c

```diff
@@ -1250,8 +1258,8 @@ devclass_alloc_unit(devclass_t dc, device_t dev, int *unitp)
 		int newsize;
 
 		oldlist = dc->devices;
-		newsize = roundup((unit + 1),
-		    MAX(1, MINALLOCSIZE / sizeof(device_t)));
+		newsize = MIN(DEVICE_UNIT_MAX, roundup((unit + 1),
+		    MAX(1, MINALLOCSIZE / sizeof(device_t))));
```

Comment:
I've redone this a little:
 https://reviews.freebsd.org/D47359
and added you as a co-author. I've also audited all the places this could have fallout, which I document in the commit message.

Also, unsigned doesn't make sense: 2^31-1 devices is plenty.

Reply:
Hopefully all the spots have been checked, but indexing an array with a signed type makes me nervous.  I'm fine with [D47359](https://reviews.freebsd.org/D47359).  This was mostly a case of during the bring-up phase I noticed this and it left me a bit uneasy.  Might not be worthy of much concern, but I did want to flag this as being a bit worrisome.

Reply:
They all have been checked. As have all the maxunit checked, and all callers.

Reply:
now this this has landed, please drop this from this series.